### DISCUSSION TITLE
v2.x.x Breaking Change to Entity History

### DISCUSSION THREAD

**Discussion by WillCodeForCats** [WillCodeForCats/solaredge-modbus-multi#21] [Announcements]
Since this integration was forked then copied from another integration, there were things from the original that don't follow the Home Assistant developer guide. One of those things is the "unique ID" for entities. Unfortunately fixing this to follow the guidelines means that entity history will be lost, although entity names will remain unchanged.

This breaking change will probably happen with a v2.0.0 release, which is to say not anytime soon, but just a heads up that it will be coming eventually.

Reference: https://developers.home-assistant.io/docs/entity_registry_index/

Edit: the future is now, v2.x.x is ready.

**Comment by purcell-lab** [comment]
@WillCodeForCats I followed your upgrade howto and have maintained my entity history.

Thanks for your work with this, it has been a great integration for my setup with two inverters and one meter.

**Reply by WillCodeForCats** [reply]
Glad to year it's working well for you. I've got more improvements coming, such as managing devices separately. Here's some sample screenshots:

<img width="309" alt="Screen Shot 2022-05-21 at 7 31 58 AM" src="https://user-images.githubusercontent.com/48533968/169656320-3387de07-d136-477b-8983-c093fafbd278.png">
<img width="1140" alt="Screen Shot 2022-05-21 at 7 34 48 AM" src="https://user-images.githubusercontent.com/48533968/169656382-9ca82927-19e3-45bd-b557-da85a939aa2e.png">
<img width="625" alt="Screen Shot 2022-05-21 at 7 31 51 AM" src="https://user-images.githubusercontent.com/48533968/169656321-423820ef-5a76-4fa7-9a18-9540848a3238.png">
<img width="1041" alt="Screen Shot 2022-05-21 at 7 36 29 AM" src="https://user-images.githubusercontent.com/48533968/169656466-1ef31e6a-a410-40ba-bf4e-5413b6ade5cb.png">

**Reply by purcell-lab** [reply]
I was going to suggest it would be good to populate the Device Info field, but it looks like you have it covered with your screenshots above.

https://developers.home-assistant.io/docs/device_registry_index

**Comment by WillCodeForCats** [comment]
I'm running 2.0.0-pre.2 on my live system and didn't lose any data after deleting, upgrading, and re-adding the integration so that it would recreate the same entity names with the fixed unique_id internally. I did end up reverting the meter entity name change because all my import/export data was lost in the energy dashboard from that change, which was unacceptable.

I will probably leave the meter naming alone. If there are multiple meters that cause a name to duplicate Home Assistant should automatically handle that by appending _2, _3, etc. to the entity names.
